elastic/kibana
# Stack monitoring - logstash 8.0 metricbeat standalone compatibility

### Summary
This change introduces the updates required to make the SM logstash section compatible with metricbeat 8.x.

- Updated query filters to target correct metricset name (`node_stats` instead of `stats`, see [sample event](https://github.com/elastic/beats/blob/master/metricbeat/module/logstash/node_stats/_meta/data.json#L135))
- Logstash stats metricsets include [a nested property defining the pipelines](https://github.com/elastic/elasticsearch/blob/master/x-pack/plugin/core/src/main/resources/monitoring-logstash.json#L251) that we aggregate over in multiple places. A nested property can't be aliased so we need to define two aggregations for each legacy and metricbeat property path ([see pipelines_nested and pipelines_nested_mb](https://github.com/elastic/kibana/issues/121346)). The implementation of the aggregation for the metricbeat path was incomplete and fixed by this change
- Updated logstash metricbeat apis and functional tests to run against the new `.monitoring-{product}-8-*` datastreams. updated the es-archives with [this script](https://gist.github.com/klacabane/873da744b436ae647b7770a1459fd1c7)

### Testing
You can find relevant informations to build each service in [this draft doc](https://github.com/elastic/observability-dev/pull/1871/files)
- start an ES service that includes [this mapping change](https://github.com/elastic/elasticsearch/pull/82152)
- start a logstash service
- start a metricbeat instance that monitors logstash and the cluster (required to load the view, logstash-only documents will load the no-data screen)
- verify data is loaded in `.monitoring-{product}-8-*`
- verify each logstash screen is loading and showing data
